index.d.ts

```diff
@@ -90,7 +90,19 @@ export interface ParseOptions {
 }
 
 export interface ParsedQuery {
-	readonly [key: string]: string | number | boolean | Array<string | number | boolean> | null | undefined;
+	[key: string]: string | Array<string> | null | undefined;
```

Comment:
you could possibly simplify this as

```typescript
export interface ParseQuery<T> {
  [key: string]: T | Array<T> | null | undefined;
}
```

and then you can just use:

```typescript
export function parse(query: string, options: { parseBooleans: true, parseNumbers: true } & ParseOptions): ParsedQuery<string | number | boolean>;
```

etc..

Reply:
Good call. I'll change it to that

Reply:
Updated. Thanks! 👍